phpBB/phpbb/console/command/user/add.php

```diff
@@ -0,0 +1,196 @@
+	* @param \phpbb\passwords\manager $password_manager The password manager used to store the user's password
+	*/
+	public function __construct(\phpbb\user $user, \phpbb\db\driver\driver_interface $db, \phpbb\config\config $config, \phpbb\passwords\manager $password_manager)
+	{
+		$this->db = $db;
+		$this->config = $config;
+		$this->password_manager = $password_manager;
+
+		$user->add_lang('ucp');
+		parent::__construct($user);
+	}
+
+	/**
+	* Sets the command name and description
+	*
+	* @return null
+	*/
+	protected function configure()
+	{
+		$this
+			->setName('user:add')
+			->setDescription($this->user->lang('CLI_DESCRIPTION_USER_ADD'))
+			->addOption('username', null, InputOption::VALUE_REQUIRED, $this->user->lang('CLI_DESCRIPTION_USER_ADD_OPTION_USERNAME'))
+			->addOption('password', null, InputOption::VALUE_REQUIRED, $this->user->lang('CLI_DESCRIPTION_USER_ADD_OPTION_PASSWORD'))
+			->addOption('email', null, InputOption::VALUE_REQUIRED, $this->user->lang('CLI_DESCRIPTION_USER_ADD_OPTION_EMAIL'))
+		;
+	}
+
+	/**
+	* Executes the command user:add
+	*
+	* If not given in option, asks the username, password and email.
+	* Then a new user is added in the database, with language and timezone found in the $config passed to the constructor, and the group_id found in the database.
+	*
+	* @param InputInterface $input The input stream used to get the options
+	* @param OutputInterface $output The output stream, used to print messages
+	*
+	* @return int 0 if all is well, 1 if a database error occured while trying to get the group_id
+	*/
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$dialog = $this->getHelperSet()->get('dialog');
+
+		$username = $input->getOption('username');
+		if (!$username) {
+			$username = $dialog->ask(
+				$output,
+				$this->user->lang('USERNAME') . $this->user->lang('COLON') . ' ',
+				null
+			);
+		}
+
+		$password = $input->getOption('password');
+		if (!$password)
+		{
+			$password = $this->get_password($output, $dialog);
+		}
+
+		$email = $input->getOption('email');
+		if (!$email)
+		{
+			$email = $dialog->ask(
+				$output,
+				$this->user->lang('EMAIL_ADDRESS') . $this->user->lang('COLON') . ' ',
+				null
+			);
+		}
+
+		try
+		{
+			$group_id = $this->get_group_id();
+		}
+		catch (\RunTimeException $e)
+		{
+			$output->writeln($e->getMessage());
+			return 1;
+		}
+
+		$user_row = array(
+			'username'				=> $username,
+			'user_password'			=> $this->password_manager->hash($password),
+			'user_email'			=> $email,
+			'group_id'				=> $group_id,
+			'user_timezone'			=> $this->config['board_timezone'],
+			'user_lang'				=> $this->config['default_lang'],
+			'user_type'				=> USER_NORMAL,
+			'user_regdate'			=> time(),
+		);
+
+		if (!function_exists('user_add'))
+		{
+			require_once dirname(__FILE__) . '/../../../../includes/functions_user.php';
+		}
+		user_add($user_row);
```

Comment:
I'm not seeing where any sort of validation is occurring when using this.  EG, for the use of a username and a possible duplicate, or email for that matter (password I guess is minutia).  Are we concerned with this?